### PR TITLE
Replace dropdowns on ask question page, profile, community selector

### PR DESCRIPTION
**Pull request by YuriyLu**

### CODE REVIEW

File: app/components/CommunitySelector/index.js [hunk 1]
```diff
@@ -1,179 +1,115 @@
...
   return (
     <Dropdown
-      isCommunitySelector
-      isArrowed={isItArrowed}
-      isOpen={isOpen}
-      toggle={toggleOpen}
-      target={
+      options={options}
+      value={value}
+      placeholder="Select a community from the list below"
```
Comment (Lampardks) [inline app/components/CommunitySelector/index.js:92]: add translate

File: app/components/CommunitySelector/index.js [hunk 2]
```diff
@@ -1,179 +1,115 @@
...
         <Button
-          communityAvatar={selectedValue?.avatar}
-          communityLabel={selectedValue?.name}
+          communityAvatar={
+            communities.find((comminity) => comminity.id === value)?.avatar
+          }
+          communityLabel={
+            communities.find((comminity) => comminity.id === value)?.name
+          }
+          placeholder={'Select a community from the list below'}
```
Comment (Lampardks) [inline app/components/CommunitySelector/index.js:105]: add translate

File: app/components/TagSelector/index.js [hunk 3]
```diff
@@ -67,14 +61,10 @@ export const TagSelector = ({
...
 }) => {
-  const [isOpen, toggleOpen] = useState(false);
-
   const [value, filteredOptions] = useMemo(() => {
     const v = (input?.value?.toJS ? input.value.toJS() : input?.value) || [];
-    const valueIds = v?.map((x) => x.id);
-
     // In menu show only which are NOT chosen
-    const fO = options?.filter((x) => !valueIds.includes(x?.id));
+    const fO = options?.filter((x) => !v.includes(x?.id));
```
Comment (Lampardks) [inline app/components/TagSelector/index.js:67]: rename v

File: app/components/TagSelector/index.js [hunk 4]
```diff
@@ -83,19 +73,23 @@ export const TagSelector = ({
     [meta],
   );
 
-  const toggle = useCallback(() => toggleOpen(!isOpen), [isOpen]);
-  const onChange = useCallback((x) => setTags([...value, x]), [setTags, value]);
+  const onChange = useCallback(
+    (x) => {
```
Comment (Lampardks) [inline app/components/TagSelector/index.js:77]: rename x

File: app/components/TagSelector/index.js [hunk 5]
```diff
@@ -105,42 +99,41 @@ export const TagSelector = ({
...
-                    <IconMd
-                      icon={closeIcon}
-                      fill={colors.tagColor || BORDER_PRIMARY}
-                      color={colors.tagColor || BORDER_PRIMARY}
-                    />
-                  </RemoveTagIcon>
-                </Tag>
-              ))}
-            </TagsContainer>
+          options={filteredOptions.map((tag) => ({
```
Comment (Lampardks) [inline app/components/TagSelector/index.js:102]: move it to a separate variable

File: app/components/TagSelector/index.js [hunk 6]
```diff
@@ -105,42 +99,41 @@ export const TagSelector = ({
...
+            label: tag.name,
+            value: tag.id,
+          }))}
+          value={value}
+          triggerClassName="full-width"
+          isMultiple={true}
+          isSearchable
+          onSelect={onChange}
+          trigger={
+            <div className="df aic">
```
Comment (Lampardks) [inline app/components/TagSelector/index.js:113]: move it to a separate component

File: app/components/common/Dropdown/Dropdown.tsx [hunk 7]
```diff
@@ -48,6 +49,15 @@ type DropdownCommonProps = {
...
   isEqualWidth?: boolean;
+  /**
+   * Is dropdown has search field
+   */
+  isSearchable?: boolean;
+  /**
+   * Class name for dropdown button
+   */
+  triggerClassName?: string;
+  appendTo?: 'viewport' | 'parent' | undefined;
```
Comment (Lampardks) [inline app/components/common/Dropdown/Dropdown.tsx:60]: delete undefined

File: app/components/common/Dropdown/Dropdown.tsx [hunk 8]
```diff
@@ -93,10 +103,13 @@ const Dropdown: React.FC<DropdownProps> = ({
   isMultiple = false,
   isDisabled = false,
   isInvalid = false,
+  isSearchable = false,
   className,
+  triggerClassName,
   onSelect,
   trigger,
   isEqualWidth,
+  appendTo,
```
Comment (Lampardks) [inline app/components/common/Dropdown/Dropdown.tsx:112]: appendTo = 'viewport'

File: app/components/common/Dropdown/Dropdown.tsx [hunk 9]
```diff
@@ -195,14 +210,41 @@ const Dropdown: React.FC<DropdownProps> = ({
...
           >
-            {items.map((option) => (
-              <DropdownOption
-                key={option.value}
-                option={option}
-                isMultiple={isMultiple}
-                onClick={onOptionClick({ close })}
-              />
-            ))}
+            {Boolean(isSearchable) && (
```
Comment (Lampardks) [inline app/components/common/Dropdown/Dropdown.tsx:213]: isSearchable

File: app/components/common/Dropdown/Dropdown.tsx [hunk 10]
```diff
@@ -195,14 +210,41 @@ const Dropdown: React.FC<DropdownProps> = ({
...
+              <div
+                css={css`
+                  padding: 5px;
+                `}
+              >
+                <Input
```
Comment (Lampardks) [inline app/components/common/Dropdown/Dropdown.tsx:219]: don't use old component, use tag input

File: app/components/common/Dropdown/Dropdown.tsx [hunk 11]
```diff
@@ -195,14 +210,41 @@ const Dropdown: React.FC<DropdownProps> = ({
...
+                    // autoFocus: true
+                  }}
+                  placeholder={'search'}
+                  type={'text'}
+                />
+              </div>
+            )}
+
+            {items
+              .filter((option) =>
```
Comment (Lampardks) [inline app/components/common/Dropdown/Dropdown.tsx:233]: move it to a separate helper function

File: app/components/common/Portal/Portal.tsx [hunk 12]
```diff
@@ -7,7 +7,7 @@ type PortalProps = {
   isFixedBody?: boolean;
 };
 
-const DEFAULT_Z_INDEX = 1000;
+const DEFAULT_Z_INDEX = 6;
```
Comment (Lampardks) [inline app/components/common/Portal/Portal.tsx:10]: don't change DEFAULT_Z_INDEX

File: app/components/common/Portal/Portal.tsx [hunk 13]
```diff
@@ -30,14 +30,14 @@ const Portal: React.FC<PortalProps> = ({
         node.classList.add('scroll-disabled');
       });
     }
-    portalAmount += 10;
+    portalAmount += 1;
```
Comment (Lampardks) [inline app/components/common/Portal/Portal.tsx:33]: don't change

File: app/components/common/Portal/Portal.tsx [hunk 14]
```diff
@@ -30,14 +30,14 @@ const Portal: React.FC<PortalProps> = ({
...
     if (portalRef.current && portalRef.current.lastElementChild) {
       const lastElementChild = portalRef.current
         .lastElementChild as HTMLElement;
       lastElementChild.style.zIndex = String(portalAmount);
     }
     return (): void => {
-      portalAmount -= 10;
+      portalAmount -= 1;
```
Comment (Lampardks) [inline app/components/common/Portal/Portal.tsx:40]: don't change

File: app/containers/Header/ProfileDropdown.jsx [hunk 15]
```diff
@@ -47,10 +47,10 @@ import LogOutIcon from 'icons/LogOut';
 const colors = singleCommunityColors();
 const styles = singleCommunityStyles();
 
-const StatusBox = styled.span`
-  display: inline-flex;
-  font-size: 14px;
-  margin-top: 2px;
+const ArrowDown = styled(Icon)`
```
Comment (Lampardks) [inline app/containers/Header/ProfileDropdown.jsx:50]: use new component

File: app/containers/Header/ProfileDropdown.jsx [hunk 16]
```diff
@@ -103,146 +103,154 @@ const B = ({ profileInfo, onClick, isMenuVisible, isMobileVersion }) => (
...
-              stroke={colors.linkColor || TEXT_PRIMARY}
-            />
-            <FormattedMessage id={messages.logout.id} />
-          </Logout>
-        </Ul>
-      </nav>
-    );
-  },
-);
+  return [
```
Comment (Lampardks) [inline app/containers/Header/ProfileDropdown.jsx:135]: put in a separate config in a separate file

File: app/containers/Header/ProfileDropdown.jsx [hunk 17]
```diff
@@ -103,146 +103,154 @@ const B = ({ profileInfo, onClick, isMenuVisible, isMobileVersion }) => (
...
+      ),
+    },
+  ];
+};
 
 const ProfileDropdown = ({ profileInfo }) => (
   <Dropdown
-    isArrowed
-    className="d-none d-md-flex"
     id="profile_dropdown_id"
```
Comment (Lampardks) [inline app/containers/Header/ProfileDropdown.jsx:243]: why id ?

File: app/containers/Header/ProfileDropdown.jsx [hunk 18]
```diff
@@ -103,146 +103,154 @@ const B = ({ profileInfo, onClick, isMenuVisible, isMobileVersion }) => (
...
-    button={<Button profileInfo={profileInfo} />}
-    menu={
-      <Menu
-        profileInfo={profileInfo}
-        questionsLength={profileInfo.postCount}
-        questionsWithUserAnswersLength={profileInfo.answersGiven}
-      />
-    }
+    options={getOptions(profileInfo).filter((option) => !!option)}
```
Comment (Lampardks) [inline app/containers/Header/ProfileDropdown.jsx:244]: add .filter((option) => !!option) to func getOptions

File: app/containers/Questions/Header.jsx [hunk 19]
```diff
@@ -76,17 +80,21 @@ const StyledCustomIconButtonContainer = styled.div`
   }
 `;
 
+const ArrowDown = styled(Icon)`
```
Comment (Lampardks) [inline app/containers/Questions/Header.jsx:83]: use new component

File: app/containers/Header/ProfileDropdown.jsx (hunk 17 above)
Comment (YuriyLu) [reply app/containers/Header/ProfileDropdown.jsx]: Removed

File: app/containers/Header/ProfileDropdown.jsx (hunk 18 above)
Comment (YuriyLu) [reply app/containers/Header/ProfileDropdown.jsx]: done

File: app/containers/Questions/Header.jsx (hunk 19 above)
Comment (YuriyLu) [reply app/containers/Questions/Header.jsx]: done

File: app/containers/Header/ProfileDropdown.jsx (hunk 16 above)
Comment (YuriyLu) [reply app/containers/Header/ProfileDropdown.jsx]: done

File: app/containers/Header/ProfileDropdown.jsx (hunk 15 above)
Comment (YuriyLu) [reply app/containers/Header/ProfileDropdown.jsx]: done

File: app/components/common/Portal/Portal.tsx (hunk 14 above)
Comment (YuriyLu) [reply app/components/common/Portal/Portal.tsx]: done

File: app/components/common/Portal/Portal.tsx (hunk 13 above)
Comment (YuriyLu) [reply app/components/common/Portal/Portal.tsx]: done

File: app/components/common/Portal/Portal.tsx (hunk 12 above)
Comment (YuriyLu) [reply app/components/common/Portal/Portal.tsx]: done

File: app/components/common/Dropdown/Dropdown.tsx (hunk 11 above)
Comment (YuriyLu) [reply app/components/common/Dropdown/Dropdown.tsx]: done

File: app/components/common/Dropdown/Dropdown.tsx (hunk 10 above)
Comment (YuriyLu) [reply app/components/common/Dropdown/Dropdown.tsx]: done

File: app/components/common/Dropdown/Dropdown.tsx (hunk 9 above)
Comment (YuriyLu) [reply app/components/common/Dropdown/Dropdown.tsx]: done

File: app/components/common/Dropdown/Dropdown.tsx (hunk 8 above)
Comment (YuriyLu) [reply app/components/common/Dropdown/Dropdown.tsx]: done

File: app/components/common/Dropdown/Dropdown.tsx (hunk 7 above)
Comment (YuriyLu) [reply app/components/common/Dropdown/Dropdown.tsx]: done

File: app/components/TagSelector/index.js (hunk 6 above)
Comment (YuriyLu) [reply app/components/TagSelector/index.js]: done

File: app/components/TagSelector/index.js (hunk 5 above)
Comment (YuriyLu) [reply app/components/TagSelector/index.js]: done

File: app/components/TagSelector/index.js (hunk 4 above)
Comment (YuriyLu) [reply app/components/TagSelector/index.js]: done

File: app/components/TagSelector/index.js (hunk 3 above)
Comment (YuriyLu) [reply app/components/TagSelector/index.js]: done

File: app/components/CommunitySelector/index.js (hunk 2 above)
Comment (YuriyLu) [reply app/components/CommunitySelector/index.js]: done

File: app/components/CommunitySelector/index.js (hunk 1 above)
Comment (YuriyLu) [reply app/components/CommunitySelector/index.js]: done

File: app/containers/Header/ProfileDropdown.jsx [hunk 20]
```diff
@@ -1,19 +1,15 @@
 /* eslint jsx-a11y/no-static-element-interactions: 0, jsx-a11y/click-events-have-key-events: 0 */
+import { ArrowDown } from 'components/icons';
```
Comment (Lampardks) [inline app/containers/Header/ProfileDropdown.jsx:2]: import ArrowDown from 'icons/ArrowDown';

File: app/containers/Header/ProfileDropdown.jsx [hunk 21]
```diff
@@ -75,174 +51,78 @@ const NoAvatarBox = styled.div`
...
-            />
-            <FormattedMessage id={messages.logout.id} />
-          </Logout>
-        </Ul>
-      </nav>
-    );
-  },
-);
+  return profileDropdownConfig(user, profileInfo, isModerator).filter(
+    (option) => !!option,
```
Comment (Lampardks) [inline app/containers/Header/ProfileDropdown.jsx:110]: (option) => Boolean(option),

File: app/containers/Questions/Header.jsx [hunk 22]
```diff
@@ -1,9 +1,11 @@
+import Icon from 'components/Icon';
+import { ArrowDown } from 'components/icons';
```
Comment (Lampardks) [inline app/containers/Questions/Header.jsx:2]: import ArrowDown from 'icons/ArrowDown';

File: app/components/common/Dropdown/Dropdown.tsx [hunk 23]
```diff
@@ -144,13 +183,24 @@ const Dropdown: React.FC<DropdownProps> = ({
...
     };
+  const [searchValue, setSearchValue] = useState();
+  const filterOptions = (options) =>
+    options.filter((option) =>
+      searchValue && typeof option.label === 'string'
+        ? option.label
+            .toLocaleLowerCase()
+            .includes(searchValue.toLocaleLowerCase())
+        : true,
+    );
```
[the remark this answers is not in the document]
Comment (YuriyLu) [reply app/components/common/Dropdown/Dropdown.tsx:194]: it desn't work

File: app/components/common/Dropdown/Dropdown.tsx [hunk 24]
```diff
@@ -86,17 +109,33 @@ type OnOptionClickParams = {
   close: PopoverContentChildrenParams['close'];
 };
 
+const InputStyled = styled.div`
```
[the remark this answers is not in the document]
Comment (YuriyLu) [reply app/components/common/Dropdown/Dropdown.tsx]: done

File: app/components/common/Dropdown/Dropdown.tsx [hunk 25]
```diff
@@ -195,7 +245,25 @@ const Dropdown: React.FC<DropdownProps> = ({
...
+              <div
+                css={css`
+                  padding: 5px;
+                `}
+              >
+                <InputStyled>
+                  <input
+                    onChange={(event: {
+                      target: { value: React.SetStateAction<undefined> };
+                    }) => setSearchValue(event.target.value)}
```
[the remark this answers is not in the document]
Comment (YuriyLu) [reply app/components/common/Dropdown/Dropdown.tsx]: done

File: app/components/CommunitySelector/index.js [hunk 26]
```diff
@@ -1,179 +1,120 @@
...
-          CustomOption={CustomOption}
-          autoFocus
-          menuIsOpen
-          isWrapped
-          placeholder={translationMessages[locale][messages.selectCommunity.id]}
-        />
-        <ManageMyCommunities />
-      </Wrapper>
-    </Dropdown>
+      zIndex={'6'}
```
[the remark this answers is not in the document]
Comment (YuriyLu) [reply app/components/CommunitySelector/index.js]: removed

File: app/containers/Header/ProfileDropdown.jsx (hunk 20 above)
Comment (YuriyLu) [reply app/containers/Header/ProfileDropdown.jsx]: done

File: app/containers/Header/ProfileDropdown.jsx (hunk 21 above)
Comment (YuriyLu) [reply app/containers/Header/ProfileDropdown.jsx]: done

File: app/components/common/Dropdown/Dropdown.tsx [hunk 27]
```diff
@@ -8,8 +8,15 @@ import {
   PopoverTriggerChildrenParams,
   PopoverContentChildrenParams,
 } from '../Popover/types';
+import ArrowDown from 'icons/ArrowDown';
 import DropdownLabel from './DropdownLabel';
 import DropdownOption from './DropdownOption';
+import InputStyled from 'components/Input/InputStyled';
```
Comment (Lampardks) [inline app/components/common/Dropdown/Dropdown.tsx:14]: don't use styled components

File: app/components/CommunitySelector/index.js (hunk 26 above)
[the remark this answers is not in the document]
Comment (Lampardks) [reply app/components/CommunitySelector/index.js:115]: not removed

File: app/components/common/Dropdown/Dropdown.tsx [hunk 28]
```diff
@@ -48,6 +55,18 @@ type DropdownCommonProps = {
...
+  /**
+   * Is dropdown has search field
+   */
+  isSearchable?: boolean;
+  /**
+   * Class name for dropdown button
+   */
+  triggerClassName?: string;
+  appendTo?: 'viewport' | 'parent';
+  zIndex?: string;
```
Comment (Lampardks) [inline app/components/common/Dropdown/Dropdown.tsx:67]: delete

File: app/components/common/Dropdown/Dropdown.tsx [hunk 29]
```diff
@@ -144,47 +169,80 @@ const Dropdown: React.FC<DropdownProps> = ({
...
+  const onChangeSearch = ({ target: { value } }) => {
+    setSearchValue(value);
+  };
 
   return (
     <Popover
       isEqualWidth={typeof isEqualWidth !== undefined ? isEqualWidth : true}
       offset={{ top: 4 }}
+      appendTo={appendTo}
+      zIndex={zIndex}
```
Comment (Lampardks) [inline app/components/common/Dropdown/Dropdown.tsx:191]: delete

**Review by Lampardks — APPROVED** [review]

**Pull request closed without merging**